_dasldemos/computational_rank.md

```diff
@@ -0,0 +1,200 @@
+
+### In your script
+
+1. [The first line should be to load the module script](https://sciwiki.fredhutch.org/scicomputing/compute_environments/#best-practice-1): `source /app/lmod/lmod/init/profile`
+2. `module load` when about to use software in your script (use specific version numbers). For example: `module load SAMtools/0.1.20-foss-2018b` 
+3. As a dry run on the command line, `module load` and check to see that you can see the executable using `which` . For example: `which samtools` should return the path where this binary is installed (`/app/software/SAMtools/0.1.20-foss-2018b/bin/samtools`).
+4. Do you work in the script. For example: `samtools view -c myfile_bam > counts.txt`
+5. `module unload` your module or `module purge` all modules when done using them and starting a new step. Try not to load all modules all at once at beginning of script. For example: `module unload SAMtools/0.1.20-foss-2018b` 
+
+Here’s an example illustrating this:
+
+## 2. Best Practices When using Containers
+
+One of the best introduction to containers is [at the Turing Way.](https://book.the-turing-way.org/reproducible-research/renv/renv-containers) There are three options to run containers: Option 1 can be directly applied in your script, wherea Option 2 requires knowledge of `WDL`. Option 3 is to use Nextflow to launch your analysis.
+
+### Containers Option 1
+
+Run software in separate isolated containers rather than a monolithic environment to avoid dependency nightmares in your bash script
+
+1. Load lmod: `source /app/lmod/lmod/init/profile`
+2. Load in `apptainer` using `module load Apptainer/1.1.6` 
+3. Use `apptainer` to `pull` a container (note that we recommend using Docker containers). For example, `apptainer pull biocontainers/samtools:v1.9-4-deb_cv1`
+4. Use Apptainer to`run` a container on your data. For example: `apptainer run biocontainers/samtools samtools view -c my_bam_file.bam > counts.txt`  will run `samtools view` and count the number of reads in your file.
+
+We recommend using `grabnode` to grab a `gizmo` node and test your script in interactive mode. This makes things a lot easier. If you don’t understand how the software is set up in the container, you can open an interactive shell into the container to test things out.
+
+### Containers Option 2
+
+Use Workflow Description Language (WDL) or Nextflow to orchestrate running containers on your data. The execution engine (such as miniWDL, Cromwell, or Nextflow) will handle loading the container as different tasks are run in your WDL script.
+
+There is a graphical user interface to running WDL scripts on your data on `gizmo` called [PROOF](https://proof.fredhutch.org/).
+
+### Containers Option 3
+
+Use Nextflow to run your software. Requires knowledge of Nextflow and Nextflow workflows.
```

Comment:
Add links to nextflow docs (e.g. https://sciwiki.fredhutch.org/hdc/hdc_workflows/)